### ISSUE TITLE
Trace does not save modified storage between different calls to the same contract

### ISSUE THREAD

**Issue by ferranbt** [trufflesuite/ganache-core#172]
Trace from `debug_traceTransaction` does not save modified storage state between different calls to the same contract.

 

## Expected Behavior

In the following contract:

```
pragma solidity ^0.4.22;

contract A {
    int i;
    
    function set(int _i) {
        i = _i;
    }
}

contract Simple {
    A a;

    function Simple() {
        a = new A();
    }

    function set() {
        a.set(1);
        a.set(2);
    }
}
```

the second time `set` is called the value of `i` (before `i = _i`) should be 1.

## Current Behavior

Using Ganache, the second time the value of `i` is 0 instead of 1. The storage field in the trace at that point is empty but it should keep the previous modified state. It works as expected with Remix and Geth.

## Steps to Reproduce (for bugs)

Run truffle debug with Ganache with the previous code and check the variable value.

**Comment by benjamincburns** [comment]
This was more than likely my oversight when adding storage variables to `debug_traceTransaction`. We'll validate this bug first just to be sure it's not a flaw in the Truffle debugger.

**Comment by eshaben** [comment]
@ferranbt, thank you for submitting this issue! The fix for this has been merged via PR #330 and will be in our next release.
